src/providers/hana/qgshanaconnectionpool.cpp

```diff
@@ -0,0 +1,55 @@
+/***************************************************************************
+   qgshanaconnectionpool.cpp
+   --------------------------------------
+   Date      : 31-05-2019
+   Copyright : (C) SAP SE
+   Author    : Maxim Rylov
+ ***************************************************************************/
+
+/***************************************************************************
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ ***************************************************************************/
+#include "qgshanaconnection.h"
+#include "qgshanaconnectionpool.h"
+
+QgsHanaConnectionPoolGroup::QgsHanaConnectionPoolGroup( const QString &name )
+  : QgsConnectionPoolGroup<QgsHanaConnection*>( name )
+{
+  initTimer( this );
+}
+
+QgsHanaConnectionPool *QgsHanaConnectionPool::sInstance = nullptr;
+
+QgsHanaConnectionPool *QgsHanaConnectionPool::instance()
+{
+  if ( !sInstance )
+    sInstance = new QgsHanaConnectionPool();
+  return sInstance;
+}
+
+bool QgsHanaConnectionPool::hasInstance()
+{
+  return sInstance != nullptr;
+}
+
+void QgsHanaConnectionPool::cleanupInstance()
+{
+  delete sInstance;
+  sInstance = nullptr;
+}
```

Comment:
Do we need these methods? What are they for?

Reply:
Done in 97d98ca.